lib/dma.c

```diff
@@ -207,11 +218,6 @@ dma_controller_remove_regions(dma_controller_t *dma)
 void
 dma_controller_destroy(dma_controller_t *dma)
 {
-    if (dma == NULL) {
```

Comment:
Why following snippet is removed?

Reply:
no need to check NULL when we only call free()